### PR TITLE
Allow specification of no external ip, boot disk size, and use existing disk.

### PR DESCRIPTION
**Pull request by rmarsagoogle**
This is the same patch I previously submitted to ansible before the modules were split.

### CODE REVIEW

**Comment by bcoca** [comment]
One thing is that when adding new features they should get a 'versions added' header in the docs, so people know what is available 'when'. 

For the options it might be nicer to allow ssd or standard, leaving the pd- prefix to be added internally.

The PR also needs a remerge.

**Comment by rmarsagoogle** [comment]
I made these changes.  Do I need to submit a new pull request?

**Comment by bcoca** [comment]
no, this one will be fine, but version added should be 1.9

if we were to publish a 1.8.3 release it would only have bugfixes, not new features.

**Comment by rmarsagoogle** [comment]
Version_added changed to 1.9.

File: cloud/google/gce.py [hunk 1]
```diff
@@ -231,7 +260,7 @@ def get_instance_info(inst):
         'name': inst.name,
         'network': netname,
         'private_ip': inst.private_ips[0],
-        'public_ip': inst.public_ips[0],
+        'public_ip': (len(inst.public_ips)>0) and inst.public_ips[0] or None,
         'status': ('status' in inst.extra) and inst.extra['status'] or None,
         'tags': ('tags' in inst.extra) and inst.extra['tags'] or [],
         'zone': ('zone' in inst.extra) and inst.extra['zone'].name or None,
```
Comment (bcoca) [inline cloud/google/gce.py:293]: avoid using ternary statements

**Comment by bcoca** [comment]
it seems like a bunch of unintended commits were added, please clean up branch

**Comment by rmarsagoogle** [comment]
Sorry, this was my first pull request and I didn't know what I was doing, so it was done from the trunk and not a branch.  I have issued a new pull request (1071) from a branch which should have everything cleaned up.

**Pull request closed without merging**